in_http: changes for lambda extension logs support

PettitWesley opened the pull request:
Signed-off-by: Wesley Pettit <wppttt@amazon.com>

<!-- Provide summary of changes -->

<!-- Issue number, if available. E.g. "Fixes #31", "Addresses #42, #77" -->

----
Enter `[N/A]` in the box, if an item is not applicable to your change.

**Testing**
Before we can approve your change; please submit the following in a comment:
- [ ] Example configuration file for the change
- [ ] Debug log output from testing the change
<!-- Invoke Fluent Bit and Valgrind as: $ valgrind ./bin/fluent-bit <args> -->
- [ ] Attached [Valgrind](https://valgrind.org/docs/manual/quick-start.html) output that shows no leaks or memory corruption was found

**Documentation**
<!-- Docs can be edited at https://github.com/fluent/fluent-bit-docs -->
- [ ] Documentation required for this feature

<!--  Doc PR (not required but highly recommended) -->

----

Fluent Bit is licensed under Apache 2.0, by submitting this pull request I understand that this code will be released under the terms of that license.

edsiper commented on plugins/in_http/http_prot.c:
```diff
@@ -76,37 +76,72 @@ int process_pack(struct flb_http *ctx, flb_sds_t tag, char *buf, size_t size)
-        if (result.data.type != MSGPACK_OBJECT_MAP) {
-            flb_plg_warn(ctx->ins, "skip record from invalid type: %i",
+        if (result.data.type == MSGPACK_OBJECT_MAP) {
+            flb_info("map!");
```
:information_source:  ?

PettitWesley replied:
oops

edsiper commented on plugins/in_http/http_prot.c:
```diff
@@ -76,37 +76,72 @@ int process_pack(struct flb_http *ctx, flb_sds_t tag, char *buf, size_t size)
+
+            break;
+    
+        } else if (result.data.type == MSGPACK_OBJECT_ARRAY) {
```
use a new line..

edsiper commented on plugins/in_http/http_prot.c:
```diff
@@ -76,37 +76,72 @@ int process_pack(struct flb_http *ctx, flb_sds_t tag, char *buf, size_t size)
+    
+        } else if (result.data.type == MSGPACK_OBJECT_ARRAY) {
+            obj = &result.data;
+            for (int i = 0; i < obj->via.array.size; i++)
```
please do not declare a variable on the ```for``` statement, that breaks with old compilers (e.g: centos 7)

edsiper commented:
once the changes are in place we can merge it.

PettitWesley commented:
@edsiper Addressed comments. I also added a new commit which adds an option tag_key which allows the user to select the tag based on a key in the record. This is also needed for Lambda support.

PettitWesley commented:
@edsiper Can we merge this?
